BDArmory/Control/MissileFire.cs

```diff
@@ -6063,7 +6063,7 @@ bool CheckEngagementEnvelope(IBDWeapon weaponCandidate, float distanceToTarget,
             //take target vel into account? //if you're going 250m/s, that's only an extra 500m to the maxRange; if the enemy is closing towards you at 250m/s, that's 250m addition
             //Max 1.5x engagement, or engagementRange + vel*4?
             //min 2x engagement, or engagement + 2000m?
-            if (!BDArmorySettings.USE_DLZ_LAUNCH_RANGE || (BDArmorySettings.USE_DLZ_LAUNCH_RANGE && weaponCandidate.GetWeaponClass() != WeaponClasses.Missile))
+            if (weaponCandidate.GetWeaponClass() != WeaponClasses.Missile || (weaponCandidate.GetWeaponClass() == WeaponClasses.Missile && ((MissileBase)weaponCandidate).UseStaticMaxLaunchRange))
```

Comment:
The `weaponCandidate.GetWeaponClass() == WeaponClasses.Missile` check is unnecessary since the logic will short-circuit and not evaluate the latter condition if the first condition is satisfied.